examples/server/tests/features/server.feature

```diff
@@ -0,0 +1,65 @@
+@llama.cpp
+Feature: llama.cpp server
+
+  Background: Server startup
+    Given a server listening on localhost:8080
+    And   a model file stories260K.gguf
+    And   a model alias tinyllama-2
+    And   42 as server seed
+    And   32 KV cache size
+    And   1 slots
+    And   32 server max tokens to predict
+    Then  the server is starting
+    Then  the server is healthy
+
+  Scenario: Health
+    Then the server is ready
+    And  all slots are idle
+
+  Scenario Outline: Completion
+    Given a prompt <prompt>
+    And   <n_predict> max tokens to predict
+    And   a completion request with no api error
+    Then  <n_predicted> tokens are predicted with content: <content>
+
+    Examples: Prompts
+      | prompt                           | n_predict | content                                                                 | n_predicted |
+      | I believe the meaning of life is | 8         | <space>going to read.                                                   | 8           |
+      | Write a joke about AI            | 64        | tion came to the park. And all his friends were very scared and did not | 32          |
+
```

Comment:
Unfortunately, we cannot test for actual generated text content because this can vary from machine to machine and from model to model. Further down the road, we can think about tests similar to the approach in https://github.com/carlini/yet-another-applied-llm-benchmark. But for now we should check only deterministic stuff

Reply:
I was thinking the `seed` fields does the job ? OK I will have a look.